components/brave_wallet/browser/ethereum_keyring.cc

```diff
@@ -138,10 +138,27 @@ bool EthereumKeyring::GetPublicKeyFromX25519_XSalsa20_Poly1305(
     return false;
   const std::vector<uint8_t> public_key =
       hd_key->GetPublicKeyFromX25519_XSalsa20_Poly1305();
-  if (!public_key.size())
+  if (public_key.empty())
     return false;
   *key = base::Base64Encode(public_key);
   return true;
 }
 
+bool EthereumKeyring::DecryptCipherFromX25519_XSalsa20_Poly1305(
+    const std::string& version,
+    const std::vector<uint8_t>& nonce,
+    const std::vector<uint8_t>& ephemeral_public_key,
+    const std::vector<uint8_t>& ciphertext,
+    const std::string& address,
+    std::vector<uint8_t>* message) {
+  HDKey* hd_key = static_cast<HDKey*>(GetHDKeyFromAddress(address));
```

Comment:
> Do have a link of where GetHDKeyFromAddress() is being defined? I've searched the codebase a few different ways to find this and haven't spotted the implementation yet.

Note that the only reason it's done this way is because that's how MetaMask defined it, so to be compatible with Dapps that use it, we need to do it the same way.  So there's no real choice here, but of course if you determined it was a security concern we would be forced to not implement it and report to MetaMask why it is so they know.

To find it just clone the brave/brave-core repo (or if you do the full steps of building the browser via brave/brave-browser go into the `src/brave dir`) and then run:

```
$ git grep GetHDKeyFromAddress

components/brave_wallet/browser/ethereum_keyring.cc:  HDKeyBase* hd_key = GetHDKeyFromAddress(address);
components/brave_wallet/browser/ethereum_keyring.cc:  HDKeyBase* hd_key = GetHDKeyFromAddress(address);
components/brave_wallet/browser/ethereum_keyring.cc:  HDKey* hd_key = static_cast<HDKey*>(GetHDKeyFromAddress(address));
components/brave_wallet/browser/ethereum_keyring.cc:  HDKey* hd_key = static_cast<HDKey*>(GetHDKeyFromAddress(address));
components/brave_wallet/browser/hd_keyring.cc:  HDKeyBase* hd_key = GetHDKeyFromAddress(address);
components/brave_wallet/browser/hd_keyring.cc:  HDKeyBase* hd_key = GetHDKeyFromAddress(address);
components/brave_wallet/browser/hd_keyring.cc:HDKeyBase* HDKeyring::GetHDKeyFromAddress(const std::string& address) {
components/brave_wallet/browser/hd_keyring.h:  HDKeyBase* GetHDKeyFromAddress(const std::string& address);
```

In this case you want the declaration from the `.h` and definition from the `.cc` one from `components/brave_wallet/browser/hd_keyring.h`

Reply:
Thanks for pointing this new code search method out. I'll add that search method to my quiver of techniques going forward. I wasn't aware git had built in `grep` functionality.

As for this issue specifically, I'll dig into this one a bit more and get back to you about my thinking and drop a comment on it ASAP.

Reply:
Discussed on slack for the way forward with this